### ISSUE TITLE
[kws_streaming] Training get slower and slower

### ISSUE THREAD

**Issue by HKAB** [google-research/google-research#1627]
Hi guys, thanks for the great work!

I'm training a KWS model using ds_tc_resnet (on NVIDIA GeForce RTX 3090, around 15 hours data). I'm currently using all augment methods include `use_spec_augment, resample, volume_resample, background_volume, background_frequency`. My model only has 70K params. I noticed that the training process became slower and slower. The first 1000 steps took 15 minutes to run, but the next 1000 steps took 25 minutes, and it became slower after that. 

As a temporary solution, I had to stopped the training, set `FLAGS.restore_checkpoint = 1`, change `FLAGS.how_many_training_steps, FLAGS.learning_rate` and start the training procedure again.

Did you guys face the same issue when training with large FLAGS.how_many_training_steps? Is it because of the IO process? I would love to have some suggestion. Thank you.

**Comment by rybakov** [comment]
Hi @HKAB I have not seen this kind of issue before. I trained all kws_streaming models on laptop with GTX 1060 6GB. RTX 3090 is much more powerful, so compute should not be a problem. I would suggest to check GPU usage and GPU memory using nvidia-smi and see how it changes in time; also check hard drive, may be there is fragmentation issue (I would assume that bottleneck is in IO)? Currently TF will not use all GPU memory with [gpu_options.allow_growth](https://github.com/google-research/google-research/blob/master/kws_streaming/train/train.py#L50)

**Comment by HKAB** [comment]
Hi @rybakov, here are some of my observations:
- [audio_processor.get_data](https://github.com/google-research/google-research/blob/master/kws_streaming/train/train.py#LL148C46-L148C70) take ~0.5s on step 2, ~1s on step 400, ~1.7s on step > 1000 and it keeps increasing. (batch size = 128)
- [model.train_on_batch](https://github.com/google-research/google-research/blob/master/kws_streaming/train/train.py#L167) take ~0.06s everytime.

I have tried:
- Read all the wav into memory instead of [read file and decode wav](https://github.com/google-research/google-research/blob/master/kws_streaming/data/input_data.py#L413), but the problem still persist, so it's not IO problem. 
- Run on CPU only, but the time loading each batch still increase, so it's not CPU to GPU problem.
- Measure the GPU time processing each batch. It isn't change, so it's not GPU problem.
- Run on different machine

As my measure, the mean time need to handle each wav file increases over time (it only takes approximately 0.005s for each wav on the first 100 steps, but increase larger than 0.01s when step > 1000), and the problem lie in [session.run() /input_data.py#L672](https://github.com/google-research/google-research/blob/master/kws_streaming/data/input_data.py#L672). Maybe running `sess.run()` over and over again make something bigger, and slower over time? Is this problem related to [this issue](https://github.com/tensorflow/tensorflow/issues/1439)? I'm not very experienced debugging Tensorflow graph, so I still can't find the core problem. Could you look more into this problem? Thank you.

**Comment by HKAB** [comment]
I close the issues since it's my fault. I accidentally added a metric which slow the training process through time.
